File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/SubstancePolymer.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class SubstancePolymer : DomainResource, IResource
{
    public Base.CodeableConcept? Class_ { get; set; }
    public SubstancePolymer_Repeat[]? Repeat { get; set; }
    public Base.CodeableConcept? Geometry { get; set; }
    public SubstancePolymer_MonomerSet[]? MonomerSet { get; set; }
    public string[]? Modification { get; set; }
    public Base.CodeableConcept[]? CopolymerConnectivity { get; set; }

    public class SubstancePolymer_Repeat_RepeatUnit_DegreeOfPolymerisation : BackboneElement
    {
        public Base.SubstanceAmount? Amount { get; set; }
        public Base.CodeableConcept? Degree { get; set; }
    }

    public class SubstancePolymer_Repeat_RepeatUnit_StructuralRepresentation : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.Attachment? Attachment { get; set; }
        public string? Representation { get; set; }
    }

    public class SubstancePolymer_Repeat_RepeatUnit : BackboneElement
    {
        public Base.SubstanceAmount? Amount { get; set; }
        public string? RepeatUnit { get; set; }
        public SubstancePolymer_Repeat_RepeatUnit_DegreeOfPolymerisation[]? DegreeOfPolymerisation { get; set; }
        public SubstancePolymer_Repeat_RepeatUnit_StructuralRepresentation[]? StructuralRepresentation { get; set; }
        public Base.CodeableConcept? OrientationOfPolymerisation { get; set; }
    }

    public class SubstancePolymer_Repeat : BackboneElement
    {
        public SubstancePolymer_Repeat_RepeatUnit[]? RepeatUnit { get; set; }
        public int? NumberOfUnits { get; set; }
        public Base.CodeableConcept? RepeatUnitAmountType { get; set; }
        public string? AverageMolecularFormula { get; set; }
    }

    public class SubstancePolymer_MonomerSet_StartingMaterial : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.SubstanceAmount? Amount { get; set; }
        public Base.CodeableConcept? Material { get; set; }
        public bool? IsDefining { get; set; }
    }

    public class SubstancePolymer_MonomerSet : BackboneElement
    {
        public Base.CodeableConcept? RatioType { get; set; }
        public SubstancePolymer_MonomerSet_StartingMaterial[]? StartingMaterial { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/SubstancePolymer.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Attachment
from base import CodeableConcept
from base import SubstanceAmount
from base import DomainResource
from base import BackboneElement

@dataclass(kw_only=True)
class SubstancePolymerRepeatRepeatUnitDegreeOfPolymerisation(BackboneElement):
    amount: Optional[SubstanceAmount] = None
    degree: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class SubstancePolymerRepeatRepeatUnitStructuralRepresentation(BackboneElement):
    type: Optional[CodeableConcept] = None
    attachment: Optional[Attachment] = None
    representation: Optional[str] = None

@dataclass(kw_only=True)
class SubstancePolymerRepeatRepeatUnit(BackboneElement):
    amount: Optional[SubstanceAmount] = None
    repeat_unit: Optional[str] = None
    degree_of_polymerisation: Optional[List[SubstancePolymerRepeatRepeatUnitDegreeOfPolymerisation]] = field(default_factory=list)
    structural_representation: Optional[List[SubstancePolymerRepeatRepeatUnitStructuralRepresentation]] = field(default_factory=list)
    orientation_of_polymerisation: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class SubstancePolymerRepeat(BackboneElement):
    repeat_unit: Optional[List[SubstancePolymerRepeatRepeatUnit]] = field(default_factory=list)
    number_of_units: Optional[int] = None
    repeat_unit_amount_type: Optional[CodeableConcept] = None
    average_molecular_formula: Optional[str] = None

@dataclass(kw_only=True)
class SubstancePolymerMonomerSetStartingMaterial(BackboneElement):
    type: Optional[CodeableConcept] = None
    amount: Optional[SubstanceAmount] = None
    material: Optional[CodeableConcept] = None
    is_defining: Optional[bool] = None

@dataclass(kw_only=True)
class SubstancePolymerMonomerSet(BackboneElement):
    ratio_type: Optional[CodeableConcept] = None
    starting_material: Optional[List[SubstancePolymerMonomerSetStartingMaterial]] = field(default_factory=list)

@dataclass(kw_only=True)
class SubstancePolymer(DomainResource):
    class_: Optional[CodeableConcept] = None
    repeat: Optional[List[SubstancePolymerRepeat]] = field(default_factory=list)
    geometry: Optional[CodeableConcept] = None
    monomer_set: Optional[List[SubstancePolymerMonomerSet]] = field(default_factory=list)
    modification: Optional[List[str]] = field(default_factory=list)
    copolymer_connectivity: Optional[List[CodeableConcept]] = field(default_factory=list)